tools/job-cache/job-cache.cpp

```diff
@@ -544,6 +859,19 @@ int main(int argc, char **argv) {
       JAST::parse(argv[3], std::cerr, job_result);
       AddJobRequest add_request(job_result);
       cache.add(add_request);
+      return 0;
+    }
+
+    if (std::string(argv[2]) == "read") {
+      JAST job_plan;
+      JAST::parse(argv[3], std::cerr, job_plan);
```

Comment:
What produces the json for the `add` operation?

Reply:
Right now me by hand, but eventually wake would do that.

That said I plan on adding a third "read or run" operation which will have a very wakebox .in.json like interface, it will invoke wakebox with a .in.json that job-cache writes, and then it will read.out.json to do the write, and it will attempt to read first before it attempts to run. That's the next step to get a demo runner working.

Reply:
I'll go work on a diagram to explain who invokes what and how now lol